drivers/video/video_stm32_venc.c

```diff
@@ -0,0 +1,1026 @@
+/*
+ * Copyright (c) 2025 Hugues Fruchet <hugues.fruchet@foss.st.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_venc
+
+#include <errno.h>
+#include <stdlib.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/irq.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/video.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/drivers/clock_control.h>
+
+#include "ewl.h"
+#include "h264encapi.h"
+#include "stm32n6xx_ll_venc.h"
+#include "reg_offset_v7.h"
+
+#include <zephyr/mem_mgmt/mem_attr_heap.h>
+#include <zephyr/dt-bindings/memory-attr/memory-attr.h>
+#include <zephyr/dt-bindings/memory-attr/memory-attr-sw.h>
+
+LOG_MODULE_REGISTER(video_stm32_venc, CONFIG_VIDEO_LOG_LEVEL);
+
+#define EWL_HEAP_ALIGNED_ALLOC(size)				\
+	mem_attr_heap_aligned_alloc((ATTR_SW_ALLOC_NON_CACHE <<	\
+				    DT_MEM_SW_ATTR_SHIFT), ALIGNMENT_INCR, size)
```

Comment:
```suggestion
	mem_attr_heap_aligned_alloc(DT_MEM_SW_ALLOC_NON_CACHE, ALIGNMENT_INCR, size)
```

Also, this uses `ALIGNMENT_INCR` which is defined later in this file.
(I know this will work but it's confusing to read)